In [30]:
import numpy as np
from matplotlib import pyplot as plt
import seaborn as sns
from pathlib import Path
from data_scripts import *
import pandas as pd


NUM_SIMS = 3
CONVERGED = False
USE_SIMULATED_TIME = False

In [31]:
data = {
    "Cold": load_data("poro_coldstart_m", n_newest=NUM_SIMS),
    "Warm": load_data("poro_warmstart_m_s", n_newest=NUM_SIMS),
}

times_dict = dict()
for name, values in data.items():
    times_dict[name] = []
    for perf in values:
        res = np.sum(
            sum_per_time_step(
                make_solve_linear_system_time(perf, converged=CONVERGED),
                perf,
                converged=CONVERGED,
            )
        )
        times_dict[name].append(res)


times = pd.DataFrame.from_dict(times_dict).T
new_data = {
    'Worst': times.max(axis=1),
    'Mean': times.mean(axis=1),
    'Best': times.min(axis=1),
}
df_mandel_m = pd.DataFrame.from_dict(new_data)
df_mandel_m

Loading data:
/home/firedrake/solver_selector/examples/2/performance/poro_coldstart_m_13.npy
/home/firedrake/solver_selector/examples/2/performance/poro_coldstart_m_14.npy
/home/firedrake/solver_selector/examples/2/performance/poro_coldstart_m_15.npy
Loading data:
/home/firedrake/solver_selector/examples/2/performance/poro_warmstart_m_s_13.npy
/home/firedrake/solver_selector/examples/2/performance/poro_warmstart_m_s_14.npy
/home/firedrake/solver_selector/examples/2/performance/poro_warmstart_m_s_15.npy


Worst       Mean       Best
Cold  65.929826  64.352032  63.181026
Warm  63.475811  63.133458  62.770948

In [32]:
data = {
    "Cold": load_data("poro_coldstart_l", n_newest=NUM_SIMS),
    # "Warmstart L(S)": load_data("poro_warmstart_l_s", n_newest=NUM_SIMS),
    # "Warmstart L(M)": load_data("poro_warmstart_l_m", n_newest=NUM_SIMS),
    "Warm": load_data("poro_warmstart_l_sm", n_newest=NUM_SIMS),
}

times_dict = dict()
for name, values in data.items():
    times_dict[name] = []
    for perf in values:
        res = np.sum(
            sum_per_time_step(
                make_solve_linear_system_time(perf, converged=CONVERGED),
                perf,
                converged=CONVERGED,
            )
        )
        times_dict[name].append(res)


times = pd.DataFrame.from_dict(times_dict).T
new_data = {
    'Worst': times.max(axis=1),
    'Mean': times.mean(axis=1),
    'Best': times.min(axis=1),
}
df_mandel_l = pd.DataFrame.from_dict(new_data)
df_mandel_l

Loading data:
/home/firedrake/solver_selector/examples/2/performance/poro_coldstart_l_13.npy
/home/firedrake/solver_selector/examples/2/performance/poro_coldstart_l_14.npy
/home/firedrake/solver_selector/examples/2/performance/poro_coldstart_l_15.npy
Loading data:
/home/firedrake/solver_selector/examples/2/performance/poro_warmstart_l_sm_13.npy
/home/firedrake/solver_selector/examples/2/performance/poro_warmstart_l_sm_14.npy
/home/firedrake/solver_selector/examples/2/performance/poro_warmstart_l_sm_15.npy


Worst        Mean        Best
Cold  201.913581  199.329702  197.256167
Warm  201.054304  199.668359  197.779866

In [33]:
data = {
    "Cold": load_data("thermal_coldstart_m", n_newest=NUM_SIMS),
    "Warm": load_data("thermal_warmstart_m_s", n_newest=NUM_SIMS),
}

times_dict = dict()
for name, values in data.items():
    times_dict[name] = []
    for perf in values:
        res = np.sum(
            sum_per_time_step(
                make_solve_linear_system_time(perf, converged=CONVERGED),
                perf,
                converged=CONVERGED,
            )
        )
        times_dict[name].append(res)


times = pd.DataFrame.from_dict(times_dict).T
new_data = {
    'Worst': times.max(axis=1),
    'Mean': times.mean(axis=1),
    'Best': times.min(axis=1),
}
df_thermal_m = pd.DataFrame.from_dict(new_data)
df_thermal_m

Loading data:
/home/firedrake/solver_selector/examples/2/performance/thermal_coldstart_m_19.npy
/home/firedrake/solver_selector/examples/2/performance/thermal_coldstart_m_20.npy
/home/firedrake/solver_selector/examples/2/performance/thermal_coldstart_m_21.npy
Loading data:
/home/firedrake/solver_selector/examples/2/performance/thermal_warmstart_m_s_13.npy
/home/firedrake/solver_selector/examples/2/performance/thermal_warmstart_m_s_14.npy
/home/firedrake/solver_selector/examples/2/performance/thermal_warmstart_m_s_15.npy


Worst       Mean       Best
Cold  95.996234  94.389007  92.424675
Warm  92.337601  88.555794  84.168483

In [34]:
data = {
    "Cold": load_data("thermal_coldstart_l", n_newest=NUM_SIMS),
    "Warm": load_data("thermal_warmstart_l_sm", n_newest=NUM_SIMS),
}

times_dict = dict()
for name, values in data.items():
    times_dict[name] = []
    for perf in values:
        res = np.sum(
            sum_per_time_step(
                make_solve_linear_system_time(perf, converged=CONVERGED),
                perf,
                converged=CONVERGED,
            )
        )
        times_dict[name].append(res)


times = pd.DataFrame.from_dict(times_dict).T
new_data = {
    'Worst': times.max(axis=1),
    'Mean': times.mean(axis=1),
    'Best': times.min(axis=1),
}
df_thermal_l = pd.DataFrame.from_dict(new_data)
df_thermal_l

Loading data:
/home/firedrake/solver_selector/examples/2/performance/thermal_coldstart_l_9.npy
/home/firedrake/solver_selector/examples/2/performance/thermal_coldstart_l_10.npy
/home/firedrake/solver_selector/examples/2/performance/thermal_coldstart_l_11.npy
Loading data:
/home/firedrake/solver_selector/examples/2/performance/thermal_warmstart_l_sm_3.npy
/home/firedrake/solver_selector/examples/2/performance/thermal_warmstart_l_sm_4.npy
/home/firedrake/solver_selector/examples/2/performance/thermal_warmstart_l_sm_5.npy


Worst         Mean         Best
Cold  2260.016742  2223.267380  2154.071525
Warm  2218.840891  2177.888141  2143.868494

In [35]:
df_m = pd.concat((df_mandel_m, df_thermal_m), join='inner', axis=1, keys=("Mandel problem", "Thermal problem"))
# df_m = df_m.style.highlight_min(axis=0, props="textbf:--rwrap;").format(precision=2)
# df_m

In [36]:
print(df_m.to_latex())

\begin{tabular}{lrrrrrr}
\toprule
 & \multicolumn{3}{r}{Mandel problem} & \multicolumn{3}{r}{Thermal problem} \\
 & Worst & Mean & Best & Worst & Mean & Best \\
\midrule
Cold & 65.929826 & 64.352032 & 63.181026 & 95.996234 & 94.389007 & 92.424675 \\
Warm & 63.475811 & 63.133458 & 62.770948 & 92.337601 & 88.555794 & 84.168483 \\
\bottomrule
\end{tabular}



In [37]:
df_l = pd.concat((df_mandel_l, df_thermal_l), join='inner', axis=1, keys=("Mandel problem", "Thermal problem"))
# df_l = df_l.style.highlight_min(axis=0, props="textbf:--rwrap;").format(precision=2)
# df_l

In [38]:
print(df_l.to_latex())

\begin{tabular}{lrrrrrr}
\toprule
 & \multicolumn{3}{r}{Mandel problem} & \multicolumn{3}{r}{Thermal problem} \\
 & Worst & Mean & Best & Worst & Mean & Best \\
\midrule
Cold & 201.913581 & 199.329702 & 197.256167 & 2260.016742 & 2223.267380 & 2154.071525 \\
Warm & 201.054304 & 199.668359 & 197.779866 & 2218.840891 & 2177.888141 & 2143.868494 \\
\bottomrule
\end{tabular}



In [39]:
df = pd.concat((df_m, df_l), keys=('M', "L"))
df = df.style.format(precision=2)
df

In [40]:
print(df.to_latex())

\begin{tabular}{llrrrrrr}
 &  & \multicolumn{3}{r}{Mandel problem} & \multicolumn{3}{r}{Thermal problem} \\
 &  & Worst & Mean & Best & Worst & Mean & Best \\
\multirow[c]{2}{*}{M} & Cold & 65.93 & 64.35 & 63.18 & 96.00 & 94.39 & 92.42 \\
 & Warm & 63.48 & 63.13 & 62.77 & 92.34 & 88.56 & 84.17 \\
\multirow[c]{2}{*}{L} & Cold & 201.91 & 199.33 & 197.26 & 2260.02 & 2223.27 & 2154.07 \\
 & Warm & 201.05 & 199.67 & 197.78 & 2218.84 & 2177.89 & 2143.87 \\
\end{tabular}



    \begin{tabular}{c | c | rrr | rrr}
    \toprule
     \multirow[c]{2}{*}{Size} & \multirow[c]{2}{*}{Start} & \multicolumn{3}{c}{Mandel problem} & \multicolumn{3}{c}{Thermal problem} \\
     &  & Worst & Mean & Best & Worst & Mean & Best \\
     \midrule

     
    \multirow[c]{2}{*}{M} & Cold & 65.93 & 64.35 & 63.18 & 96.00 & 94.39 & 92.42 \\
     & Warm & 63.48 & 63.13 & 62.77 & 92.34 & 88.56 & 84.17 \\
     \midrule
    \multirow[c]{2}{*}{L} & Cold & 201.91 & 199.33 & 197.26 & 2260.02 & 2223.27 & 2154.07 \\
     & Warm & 201.05 & 199.67 & 197.78 & 2218.84 & 2177.89 & 2143.87 \\
    
    \bottomrule
    \end{tabular}